src/batch.js

```diff
@@ -42,26 +41,22 @@ exports.processSortedSet = async function (setKey, process, options) {
 		process = util.promisify(process);
 	}
 
-	await async.whilst(
-		function (next) {
-			next(null, !done);
-		},
-		async function () {
-			const ids = await db['getSortedSetRange' + (options.withScores ? 'WithScores' : '')](setKey, start, stop);
-			if (!ids.length || options.doneIf(start, stop, ids)) {
-				done = true;
-				return;
-			}
-			await process(ids);
-
-			start += utils.isNumber(options.alwaysStartAt) ? options.alwaysStartAt : options.batch + 1;
-			stop = start + options.batch;
-
-			if (options.interval) {
-				await sleep(options.interval);
-			}
+	while (!done) {
+		/* eslint-disable no-await-in-loop */
+		const ids = await db['getSortedSetRange' + (options.withScores ? 'WithScores' : '')](setKey, start, stop);
+		if (!ids.length || options.doneIf(start, stop, ids)) {
+			done = true;
```

Comment:
There are a few occurrences of replacing async.whilst where you don't need the `done` variable at all, since you either `break` or `return` directly after setting it. Meaning you can just use a `while(true) { ... }` loop instead.

Reply:
eslint complains about this as well `while (true) {` :trollface:

Reply:
Should be able to use `for(;;) { ... }`, or set the checkloops option in the eslint config: https://eslint.org/docs/rules/no-constant-condition#checkloops